policy/release/trusted_task.rego

```diff
@@ -91,6 +91,33 @@ deny contains result if {
 	result := lib.result_helper_with_term(rego.metadata.chain(), [err.msg], err.term)
 }
 
+# METADATA
+# title: Test Tasks are trusted
+# description: >-
+#   Check the trust of the Tekton Tasks used for testing in the build Pipeline. This rule is used if 
+#   Trusted Artifacts are enabled. In this case, a chain of trust is established for all the Tasks 
+#   involved in creating an artifact. If the chain contains an untrusted Task, then a violation is emitted. 
+# custom:
+#   short_name: trusted
+#   failure_msg: "%s"
+#   solution: >-
+#     If using Trusted Artifacts, be sure every Task in the build Pipeline responsible for producing
+#     a Trusted Artifact is trusted. Otherwise, ensure **all** Tasks in the build Pipeline are
+#     trusted. Note that trust is eventually revoked from Tasks when newer versions are made
+#     available.
+#   collections:
+#   - redhat
+#   effective_on: 2024-07-12T00:00:00Z
+#
+deny contains result if {
+	_uses_trusted_artifacts
+	some attestation in lib.pipelinerun_attestations
+	some test_task in tkn.tasks_test_output(attestation)
```

Comment:
I don't think we want to add a new set of rules here. We probably just want to rework the existing rules to pick up the builder and the test tasks.

Reply:
https://github.com/enterprise-contract/ec-policies/pull/1040/commits/6bbee2166833b472245d8b76577ec175decd853e